src/android/FileTransferBackground.java

```diff
@@ -52,6 +56,9 @@
     private static String currentTag;
     private static long currentTagFetchedAt;
 
+    private ExecutorService executorService = null;
+    private ScheduledExecutorService scheduledExecutorService = null;
```

Comment:
Can use only 1 and set delay 0 for some. Initial events can be sent at fixed intervals.